packages/tokamak/sdk/tasks/deposit-withdraw-eth.ts

```diff
@@ -0,0 +1,264 @@
+      'OptimismPortalProxy'
+    )
+    optimismPortal = Deployment__OptimismPortal.address
+  }
+
+  if (l2OutputOracle === '') {
+    const Deployment__L2OutputOracle = await hre.deployments.get(
+      'L2OutputOracleProxy'
+    )
+    l2OutputOracle = Deployment__L2OutputOracle.address
+  }
+}
+
+const depositETH = async (amount: NumberLike) => {
+  console.log('Deposit TON:', amount)
+  console.log('l1 address:', l1Wallet.address)
+  console.log('l2 address:', l2Wallet.address)
+
+  const ethContract = new ethers.Contract(predeploys.ETH, erc20ABI, l2Wallet)
+
+  const l1Contracts = {
+    StateCommitmentChain: zeroAddr,
+    CanonicalTransactionChain: zeroAddr,
+    BondManager: zeroAddr,
+    AddressManager: addressManager,
+    L1CrossDomainMessenger: l1CrossDomainMessenger,
+    L1StandardBridge: l1StandardBridge,
+    OptimismPortal: optimismPortal,
+    L2OutputOracle: l2OutputOracle,
+  }
+  console.log('l1 contracts:', l1Contracts)
+
+  const bridges = {
+    TON: {
+      l1Bridge: l1Contracts.L1StandardBridge,
+      l2Bridge: predeploys.L2StandardBridge,
+      Adapter: TONBridgeAdapter,
+    },
+  }
+
+  const l1ChainId = (await l1Provider.getNetwork()).chainId
+  const l2ChainId = (await l2Provider.getNetwork()).chainId
+
+  const messenger = new CrossChainMessenger({
+    bedrock: true,
+    contracts: {
+      l1: l1Contracts,
+    },
+    bridges,
+    l1ChainId,
+    l2ChainId,
+    l1SignerOrProvider: l1Wallet,
+    l2SignerOrProvider: l2Wallet,
+  })
+
+  let l1Balance = await l1Wallet.getBalance()
+  console.log('l1 eth balance: ', l1Balance.toString())
+
+  let l2Balance = await ethContract.balanceOf(l2Wallet.address)
+  console.log('l2 eth balance:', l2Balance.toString())
+
+  const depositTx = await messenger.depositETH(amount)
+  await depositTx.wait()
+  console.log('depositTx:', depositTx.hash)
+
+  await messenger.waitForMessageStatus(depositTx.hash, MessageStatus.RELAYED)
+
+  l1Balance = await l1Wallet.getBalance()
+  console.log('l1 eth balance: ', l1Balance.toString())
+
+  l2Balance = await ethContract.balanceOf(l2Wallet.address)
+  console.log('l2 eth balance:', l2Balance.toString())
+}
+
+const withdrawETH = async (amount: NumberLike) => {
+  console.log('Withdraw ETH:', amount)
+  console.log('l1 address:', l1Wallet.address)
+  console.log('l2 address:', l2Wallet.address)
+
+  const ethContract = new ethers.Contract(predeploys.ETH, erc20ABI, l2Wallet)
+
+  const l1Contracts = {
+    StateCommitmentChain: zeroAddr,
+    CanonicalTransactionChain: zeroAddr,
+    BondManager: zeroAddr,
+    AddressManager: addressManager,
+    L1CrossDomainMessenger: l1CrossDomainMessenger,
+    L1StandardBridge: l1StandardBridge,
+    OptimismPortal: optimismPortal,
+    L2OutputOracle: l2OutputOracle,
+  }
+
+  const bridges = {
+    TON: {
+      l1Bridge: l1Contracts.L1StandardBridge,
+      l2Bridge: predeploys.L2StandardBridge,
+      Adapter: TONBridgeAdapter,
+    },
+  }
```

Comment:
```suggestion
  const bridges = {
    ETH: {
      l1Bridge: l1Contracts.L1StandardBridge,
      l2Bridge: predeploys.L2StandardBridge,
      Adapter: ETHBridgeAdapter,
    },
  }
```

Reply:
I think ETHBridgeAdapter is for legacy usage, I made TONBridgeAdapter for the new feature

Reply:
I'm sorry, I'm having trouble understanding your idea. Can you please provide more information or clarify your point?
(I updated `ETHBridgeAdapater` on my processing PR, that's why I suggest use `ETHBridgeAdapter`)

Reply:
Oh, I did not know your PR when creating this PR.
I saw your PR, I think we can use withdrawETH after merging your PR. 

Actually, in L2, ETH is an ERC20 so that I am withdrawing ETH by using withdrawERC20 and TON by using withdrawETH. But it is a bit confuse when using withdrawETH for TON on L2 although other teams don't need to update their code according the changed SDK. 

What do you guy think? @tokamak-network/layer2-core ?

Reply:
I think still works well, but there seems to be some confusion when we use token names interchangeably 😄